downloads/swan-lake-release-notes/swan-lake-beta3/RELEASE_NOTE.md

```diff
@@ -0,0 +1,780 @@
+    service class Bar {
+    *Foo;
+    
+    function f2() returns int => 36;
+    }
+    ```
+
+- A bug in string unescaping of Unicode codepoints > `0xFFFF` has been fixed.
+    ```ballerina
+    import ballerina/io;
+    
+    public function main() {
+    string str = "Hello world! \u{1F600}";
+    io:println(str);
+    }
+    ```
+
+The above code snippet, which previously printed `Hello world!  ὠ0` will now print `Hello world! 😀`.
+
+- A bug in escaping `NumericEscape` has been fixed.
+    ```ballerina
+    import ballerina/io;
+    
+    public function main() {
+    string str = "\\u{61}pple";
+    io:println(str);
+    }
+    ```
+
+This code snippet, which previously printed `\u0061pple` will now print `\u{61}pple`.
+
+- A bug that resulted in `NumericEscape` in the template string not being interpreted literally has been fixed.
+    ```ballerina
+    import ballerina/io;
+    
+    public function main() {
+    string str = string `\u{61}pple`;
+    io:println(str);
+    }
+    ```
+
+The code snippet above, which previously printed `\u0061pple` will now print `\u{61}pple`.
+
+- A bug that resulted in self-referencing not being detected when referenced via a `let` expression or a constant reference expression has been fixed.
+
+    The following will now result in errors.
+    ```ballerina
+    const int INTEGER = INTEGER; // Compilation error.
+
+    public function main() {
+        string s = let string[] arr = [s] in arr[0]; // Compilation error.
+    }
+    ```
+
+- Type narrowing is now reset on any assignment to a narrowed variable. Previously, type narrowing was not reset if the static type of the new value was a subtype of the narrowed type. This was a deviation from the specification.
+    ```ballerina
+    public function main() {
+        int|string v = 0;
+
+        if v is int {
+            int x = v; // Valid.
+
+            // Now, the type is reset to `int|string` even though `1` belongs to `int`.
+            v += 1;
+
+            int y = v; // Invalid, will now result in an error.
+        }
+    }
+    ```
+
+To view all bug fixes, see the [GitHub milestone for Swan Lake Beta3](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+Beta3%22+label%3AType%2FBug+label%3ATeam%2FCompilerFE).
+
+## Runtime Updates
+
+### Improvements
+
+#### Improved Configurable Variables to Support XML Types Through TOML Syntax
+
+The `configurable` feature is improved to support variables with XML types through the TOML syntax.
+
+For example, if the XML-typed configurable variables are defined in the following way,
+
+``` ballerina
+configurable xml xmlVar = ?;
+```
+the values can be provided in the `Config.toml` file as follows.
+
+
+```toml
+xmlVar = "<book><name>Sherlock Holmes</name></book>"
+```
+
+#### Improved Configurable Variables to Support Additional Fields and Rest Fields for Records
```

Comment:
Not sure I get what additional fields means here? @HindujaB @warunalakshitha 

Shouldn't this be
```suggestion
#### Improved Configurable Variables to Support Open Records
```

Reply:
Configurability was supported earlier for open records if we don't configure any extra fields. An improvement is done to support extra fields shapes in record values. 
I think we can use the term `extra field shapes` instead of  both `Additional Fields` and `Rest Fields` , as it is used in the spec.

Reply:
How about
```suggestion
#### Improved Configurable Variables to Support Extra Fields with Open Records
```

or
```suggestion
#### Support to Provide Extra Fields for Configurable Variables of Open Record Types
```

We'll have to update the description too.

Reply:
@HindujaB both the following are open records and only open records allow extra fields.

```ballerina
type Foo record {|
    anydata...;  
|};

type Bar record {
};
```

Didn't get what you mean by "I think we can remove the open record part, as rest fields are not supported for open records."?